sumo-api/sumo/threedee/model_sampler.pyx

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+import numpy as np
+
+class ModelSampler(object):
+    """
+    Algorithm to sample points on a GltfModel.
+    """
+
+    def __init__(self, sampling_density=625):
+        self._sampling_density = sampling_density
+
+    def run(self, model):
+        """
+        Randomly sample points from the faces of <model> and return
+        the resulting points.  Colors are interpolated from the vertex
+        colors of the sampled face.
+
+        Inputs:
+        model (GltfModel) - model to sample
+
+        Return:
+        points (N x 6 numpy array) - N points by 6 (x,y,z,R,G,B)
+        """
+
+        # count the # of faces in all the component meshes
+        # and allocate space for faces
+        counts = [m.num_indices() for m in model.primitive_meshes()]
+        n_faces3 = np.sum(counts)  # 3 x num faces
+        faces = np.zeros((n_faces3, 6))
+
+        # build faces list from element component meshes
+        start = 0
+
+        for i, mesh in enumerate(model.primitive_meshes()):
+            # replicate verts based on face ids
+            # temp[:,i] = vertex for face_id i
```

Comment:
Remove

Reply:
this is not code.  It is a comment to explain the structure